javascript/modules/client/src/internal/modules/ipfs.ts

```diff
@@ -0,0 +1,92 @@
+import { Random } from "@aragon/sdk-common";
+import { Context } from "../../context";
+import { IClientIpfsCore } from "../interfaces/core";
+import { Client as IpfsClient } from "@aragon/sdk-ipfs";
+
+export class IPFSModule implements IClientIpfsCore {
+    private _ipfs: IpfsClient[] = [];
+    private _ipfsIdx: number = -1;
+
+    constructor(context: Context) {
+        if (context.ipfs?.length) {
+            this._ipfs = context.ipfs;
+            this._ipfsIdx = Math.floor(Random.getFloat() * context.ipfs.length);
+        }
+    }
+
+    public getClient(): IpfsClient {
+        if (!this._ipfs[this._ipfsIdx]) {
+            throw new Error("No IPFS endpoints available");
+        }
+        return this._ipfs[this._ipfsIdx];
+    }
+
+    /**
+     * Starts using the next available IPFS endpoint
+     */
+    public shiftClient(): void {
+        if (!this._ipfs?.length) {
+            throw new Error("No IPFS endpoints available");
+        } else if (this._ipfs?.length < 2) {
+            throw new Error("No other endpoints");
+        }
+        this._ipfsIdx = (this._ipfsIdx + 1) % this._ipfs?.length;
+    }
+
+    /** Returns `true` if the current client is on line */
+    public async isUp(): Promise<boolean> {
+        if (!this._ipfs?.length) return Promise.resolve(false);
+
+        try {
+            await this.getClient().nodeInfo()
+            return true
+        } catch (e) {
+            return false
+        }
```

Comment:
Same as before, an `async/await` function has a heavier footprint, is harder to debug and with the given logic could be kept as a plain function returning a chain of promises